src/screen.cpp

```diff
@@ -434,52 +431,58 @@ void IRAM_ATTR ScreenUpdateLoopEntry(void *)
 
-            // When the button is pressed advance to the next information page on the little display
+                // When the button is pressed advance to the next information page on the little display
 
-            g_InfoPage = (g_InfoPage + 1) % g_InfoPageCount;
+                g_InfoPage = (g_InfoPage + 1) % g_InfoPageCount;
 
-            auto& effectManager = g_ptrSystem->EffectManager();
+                auto& effectManager = g_ptrSystem->EffectManager();
 
-            // We stop rotating the effects when we are on the debug info page, and resume when we are not
-            if (g_InfoPage == 0)
-            {
-                effectInterval = effectManager.GetInterval();
-                effectManager.SetInterval(0, true);
-            }
-            // Restore effect interval to the value we remembered, on the proviso that effect rotation is now indeed
-            // paused. Otherwise, the user may have chosen a different effect interval while we weren't looking and
-            // we don't want to mess with that.
-            else if (effectManager.GetInterval() == 0)
-                effectManager.SetInterval(effectInterval, true);
+                // We stop rotating the effects when we are on the debug info page, and resume when we are not
+                if (g_InfoPage == 0)
+                {
+                    effectInterval = effectManager.GetInterval();
+                    effectManager.SetInterval(0, true);
+                }
+                // Restore effect interval to the value we remembered, on the proviso that effect rotation is now indeed
+                // paused. Otherwise, the user may have chosen a different effect interval while we weren't looking and
+                // we don't want to mess with that.
+                else if (effectManager.GetInterval() == 0)
+                    effectManager.SetInterval(effectInterval, true);
 
-            bRedraw = true;
-        }
-#endif
+                bRedraw = true;
+            }
+        #endif
 
-#ifdef TOGGLE_BUTTON_2
-        Button2.update();
-        if (Button2.pressed())
-        {
-            debugI("Button 2 pressed on pin %d so advancing to next effect", TOGGLE_BUTTON_2);
-            g_ptrSystem->EffectManager().NextEffect();
-            bRedraw = true;
-        }
-#endif
+        #ifdef TOGGLE_BUTTON_2
+            Button2.update();
+            if (Button2.pressed())
+            {
+                debugI("Button 2 pressed on pin %d so advancing to next effect", TOGGLE_BUTTON_2);
+                g_ptrSystem->EffectManager().NextEffect();
+                bRedraw = true;
+            }
+        #endif
 
         UpdateScreen(bRedraw);
         if (g_Values.UpdateStarted)
+        {
             delay(200);
+        }
         else
-            delay(50);
-
+        {
+            #if AMOLED_S3
+                lv_task_handler();
+            #endif
+            delay(2);
```

Comment:
Are you sure you want to decrease the delay to 2ms for all screens? Not just in case the AMOLED_S3 handler is called?